portal-web/test/functional/com/liferay/portalweb/macros/MultiFactorAuthentication.macro

```diff
@@ -228,6 +253,37 @@ definition {
 			value1 = "Your one-time password has been sent by email.");
 	}
 
+	macro verifyCustomOTP {
+		if (isSet(specificURL)) {
+			Navigator.openSpecificURL(url = "${specificURL}");
+		}
+		else {
+			Navigator.openURL();
+		}
+
+		SignIn._signIn(
+			userEmailAddress = "${userEmailAddress}",
+			userPassword = "${userPassword}");
+
+		AssertElementPresent(locator1 = "MultiFactorAuthentication#CUSTOM_MFA_TITLE");
+		AssertElementPresent(locator1 = "MultiFactorAuthentication#CUSTOM_MFA_DESCRIPTION");
+		AssertElementPresent(locator1 = "MultiFactorAuthentication#EMAIL_OTP_LINK");
+
+		Type(
+			locator1 = "MultiFactorAuthentication#CUSTOM_MFA_VERIFY_FIELD",
+			value1 = "${verificationText}");
+
+		Button.click(button = "Submit");
+
+		if (IsElementPresent(locator1 = "MultiFactorAuthentication#EMAIL_OTP_VERIFY_FAILED_MESSAGE")) {
+			AssertElementNotPresent(locator1 = "UserBar#USER_AVATAR_IMAGE");
+		}
+		else {
+			AssertElementNotPresent(locator1 = "MultiFactorAuthentication#EMAIL_OTP_VERIFY_FIELD");
+			AssertElementPresent(locator1 = "UserBar#USER_AVATAR_IMAGE");
+		}
```

Comment:
We should explicitly tell this macro what to check for. If the test were to fail here for whatever reason when we were expecting`EMAIL_OTP_VERIFY_FIELD`, we would not catch the failure at this step since the `USER_AVATAR_IMAGE` expectedly wouldn't show up.

Reply:
I would review verityOTP step with @dellawang, I think we shall split the assertion based on the various verifiers MFA support. https://issues.liferay.com/browse/LRQA-63371